renderdoc/serialise/codecs/vk_cpp_codec/ext_object.h

```diff
@@ -0,0 +1,190 @@
+  // would not be able to distibguish it from any other Vk*Flags and it would lose
+  // the ability to stringify the bit mask into a human readable string.
+  else if(ptr->type.name == "VkPipelineStageFlagBits")
+    return pipeline_stage_flags.c_str();
+
+  return ptr->type.name.c_str();
+}
+
+inline std::string ValueStr(SDObject *ptr)
+{
+  RDCASSERT(ptr->IsSimpleType());
+  std::string result;
+
+  if(ptr->IsBuffer())
+  {
+    std::string buf_name = ptr->AsString();
+    RDCASSERT(!buf_name.empty());
+    // A value for for a Buffer is it's $name.data().
+    result = buf_name + ".data()";
+    // just-in-time fix for vkCreatShaderModule pCode variable
+    if(ptr->name == "pCode")
+      result = "(const uint32_t*) " + result;
+  }
+  else if(ptr->IsNULL())
+  {
+    result = "NULL";
+  }
+  else if(ptr->IsUInt())
+  {
+    result = std::to_string(ptr->AsUInt64()) + "u";
+  }
+  else if(ptr->IsInt())
+  {
+    result = std::to_string(ptr->AsInt64());
+  }
+  else if(ptr->IsFloat())
+  {
+    if(isnan(ptr->data.basic.d))
+      ptr->data.basic.d = 1.0f;
+    result = std::to_string(ptr->AsDouble()) + "f";
+  }
+  else if(ptr->IsEnum())
+  {
+    result = ptr->data.str;
+  }
+  else if(ptr->IsString())
+  {
+    std::string escaped;
+    escaped.reserve(ptr->data.str.size());
+    for(char c : ptr->data.str)
+    {
+      switch(c)
+      {
+        case '\a': escaped += "\\a"; break;
+        case '\b': escaped += "\\b"; break;
+        case '\f': escaped += "\\f"; break;
+        case '\n': escaped += "\\n"; break;
+        case '\r': escaped += "\\r"; break;
+        case '\t': escaped += "\\t"; break;
+        case '\v': escaped += "\\v"; break;
+
+        case '"':
+        case '\\':
+          escaped.push_back('\\');
+          escaped.push_back(c);
+          break;
+
+        default:
+          if(c < 32 || c > 127)
+          {
+            char buf[8] = {};
+            snprintf(buf, sizeof(buf), "\\x%02X", c);
+            escaped += buf;
+          }
+          else
+          {
+            escaped.push_back(c);
+          }
+          break;
+      }
+    }
+    result = std::string("\"") + escaped + std::string("\"");
+  }
+  return result;
+}
+
+inline uint32_t ChunkID(const SDObject *ptr)
+{
+  RDCASSERT(ptr->type.basetype == SDBasic::Chunk);
+  const SDChunk *chunk = (const SDChunk *)(ptr);
+  return chunk->metadata.chunkID;
+}
+
+inline uint64_t CanonicalUnionBranch(SDObject *ptr)
+{
+  if(ptr->type.name == "VkClearValue")
+  {
+    return 0;    // Use `color`
+  }
+  else if(ptr->type.name == "VkClearColorValue")
+  {
+    return 2;    // Use `uint32`
+  }
+  // Attempting to output an unknown union type.
+  // This function must be modified to return the index of the canonical branch, which should be
+  // chosen so that it's size is equal to the size of the entire union, and so that the values can
+  // be represented exactly (e.g., not floating point).
+  RDCASSERT(0);
```

Comment:
Please prefer a descriptive error message with `RDCERR` over just `RDCASSERT(0)`. when there's no actual condition to check.

Reply:
Done.